### PR TITLE
Accept incoming p2p connections

### PR DESCRIPTION
**Pull request by ixje**
**What current issue(s) does this address, or what feature is it adding?**
FIx https://github.com/CityOfZion/neo-python/issues/522

**How did you solve this problem?**
Add endpoint for incoming connections, implemented first part of `RemoteNode.StartProtocol()` routine up to `verack` [checking](https://github.com/neo-project/neo/blob/master/neo/Network/RemoteNode.cs#L442)

**How did you make sure your solution works?**
manual testing

**Are there any special changes in the code that we should be aware of?**
Yes. I'm aware that our NeoNode is implemented differently that the C#'s equivalent `RemoveNode` due to historical performance issues. Due to the way it's structured now I had to squeeze in the `Version` and `Verack` payload sending in different locations unlike the `StartProtocol` routine in C#. I also did not go as far as implementing the sequence for `getheaders` ([here](https://github.com/neo-project/neo/blob/master/neo/Network/RemoteNode.cs#L447))  and `getblocks` as we already can update our own state using the peers that we connect to. It would keep the code a bit simpler. Interested to hear if this should still be added

**Please check the following, if applicable:**

- [ ] Did you add any tests?
- [X] Did you run `make lint`?
- [X] Did you run `make test`?
- [X] Are you making a PR to a feature branch or development rather than master?
- [X] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

**Comment by ixje** [comment]
don't merge yet. I spotted a missed commit and might be on to a bug. investigating now

**Comment by ixje** [comment]
For what it's worth; I wrote a small test program that completes the handshake against a C# client, then did the same against `neo-python` with this PR and that now passes after the above fixes.

**Comment by ixje** [comment]
@localhuman let me know what insights you want for this PR to pass. I could share the test program on discord if that helps in building confidence.

File: neo/bin/prompt.py [hunk 1]
```diff
@@ -1121,6 +1122,9 @@ def main():
     reactor.callInThread(cli.run)
     NodeLeader.Instance().Start()
 
+    endpoint = TCP4ServerEndpoint(reactor, settings.NODE_PORT)
+    endpoint.listen(NeoClientFactory(incoming_client=True))
+
```
[the remark this answers is not in the document]
Comment (ixje) [reply neo/bin/prompt.py]: Good point, will move

**Comment by ixje** [comment]
It's not clear to me if you want me to validate other possible incoming requests in this PR or not

**Comment by ixje** [comment]
fyi; working on it. There are indeed some missing requests and requests that get handled incorrectly.

**Comment by ixje** [comment]
So I tested this as follows;

- run a private net
- let `neo-python` sync, then exit
- disable connecting to seeds in `neo-python` and run instance again to avoid C# client grabbing blocks via connected seeds
- set C# client seedlist to `neo-python` instance only (same reason as above)
- let C# client sync (=is successful)
- compare blocks from `neo-python` with C# RPC output (they're equal)

If you want to test it yourself you might need to increase the `CONNECTED_PEER_MAX` from 5 to 6 or higher because for some reason there's always 1 privatenet node that's connected twice and therefore you'll get a connection rejected. 
https://github.com/CityOfZion/neo-python/blob/ba815b870614d01f9295d934c8dc3a172765e1aa/neo/Network/NodeLeader.py#L172
The alternative is commenting out the code for connecting to the seeds
https://github.com/CityOfZion/neo-python/blob/ba815b870614d01f9295d934c8dc3a172765e1aa/neo/Network/NodeLeader.py#L110-L114

**Comment by localhuman** [comment]
Looks good! Going to try it out for a day or two and see how it goes

**Comment by localhuman** [comment]
Seems pretty good after running for a while.  One thing, could you turn off `print_payload=True` 😉

**Comment by localhuman** [comment]
Looks good! just need to fix the changelog and we're ready to go 🎉

**Pull request merged**